lib/travis_dedup.rb

```diff
@@ -98,17 +100,36 @@ def host
     end
 
     def request(method, path, params, headers={})
+      attempts = 0
+      begin
+        faraday_send(method, path, params, headers)
+      rescue TravisError500 => error
+        raise error if (attempts += 1) > max_attempts
+        retry
+      end
+    end
+
+    def max_attempts
+      DEFAULT_MAX_ATTEMPTS_WHEN_TRAVIS_DOES_NOT_RESPOND
+    end
+
+    class TravisError500 < Faraday::Error; end
+
+    def faraday_send(method, path, params, headers={})
       response = Faraday.send(method, "#{host}/#{path}", params, headers)
       case response.status
       when 200 then JSON.parse(response.body)
       when 204 then nil
+      when 500
```

Comment:
I will check Faraday code. Judging by the stackstrace (see https://github.com/grosser/travis_dedup/issues/9 ), I would say it does not raise on 500